src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,75 @@
+package duke;
+
+import duke.command.ByeCommand;
+import duke.command.Command;
+import duke.command.DeadlineCommand;
+import duke.command.DeleteCommand;
+import duke.command.EventCommand;
+import duke.command.ListCommand;
+import duke.command.MarkCommand;
+import duke.command.TodoCommand;
+import duke.command.UnmarkCommand;
+
+/**
+ * A class to parse commands received from user input.
+ */
+public class Parser {
+    /** Check for program exit (true when bye command entered).*/
+    private boolean isExit;
+
+    /**
+     * Constructor for Parser class.
+     */
+    public Parser() {
+        isExit = false;
+    }
+
+    /**
+     * Parses a command received from input.
+     * @param commandStr command to be parsed.
+     * @return The Command obtained from parsing.
+     * @throws DukeException If command input is invalid.
+     */
+    public Command parse(String commandStr) throws DukeException{
+        switch (commandStr.split("\\s+")[0]) {
+        case "list":
+            return new ListCommand();
+        case "todo":
+            return new TodoCommand();
+        case "deadline":
+            return new DeadlineCommand();
+        case "event":
+            return new EventCommand();
+        case "mark":
+            return new MarkCommand();
+        case "unmark":
+            return new UnmarkCommand();
+        case "delete":
+            return new DeleteCommand();
+        case "bye":
+            isExit = true;
+            return new ByeCommand();
+        default:
+            throw new DukeException("Invalid command");
+        }
+    }
+
+    /**
+     * Utility function to find index of element in a String array.
+     * @param arr The array to search.
+     * @param item The item to find.
+     * @return Index of item if present; -1 otherwise.
+     */
```

Comment:
I like the effort in giving every class and function a javadoc. However, should the description be in this form?